### PR TITLE
Milestone 5

### PR DESCRIPTION
**Pull request by AbigiyaTY**
In this milestone I:

- Write tests for controllers.
image

### CODE REVIEW

**Review by prolajumokeoni — COMMENTED** [review]
Hi @AbigiyaTY 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have all headers present in your readme, Make sure to add the section Future Features, including 1-3 potential features to be added,
kindly ensure all main headers from the [README template](https://github.com/microverseinc/readme-template/blob/master/README.md) are present 👍🏿 


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by prolajumokeoni — COMMENTED** [review]
Hi @AbigiyaTY 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have all headers present in your readme, Make sure to add the section Future Features, including 1-3 potential features to be added,
kindly ensure all main headers from the [README template](https://github.com/microverseinc/readme-template/blob/master/README.md) are present 👍🏿 


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'users#show' do
+    it 'show for users works!' do
+      get '/users/1'
+      expect(response).to have_http_status(:ok)
+      expect(response.body).to include('Teacher from Mexico.')
+    end
```
Comment (mikethreels) [inline spec/controllers/users_controller_spec.rb]: - [ ] your spec file should include tests that ensure the correct view template is rendered.
Example : `expect(response).to render_template(:show)`

File: spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :request do
+  describe 'GET #index' do
+    it 'returns a success response' do
+      get users_url
+      expect(response).to be_successful
+    end
+  end
```
Comment (mikethreels) [inline spec/controllers/users_controller_spec.rb]: - [ ] your spec file should include tests that ensure the correct view template is rendered.
Example : `expect(response).to render_template(:index)`

File: spec/controllers/posts_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      expect(response.body).to include('Posts')
+    end
+  end
+
+  describe 'posts#show' do
+    it 'show for post works!' do
+      get '/users/1/posts/1'
+      expect(response).to have_http_status(:ok)
+      expect(response.body).to include('This is my first post')
+    end
```
Comment (mikethreels) [inline spec/controllers/posts_controller_spec.rb]: - [ ] your spec file should include tests that ensure the correct view template is rendered.
Example : `expect(response).to render_template(:show)`

File: spec/controllers/posts_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /index' do
+    it 'index for post works!' do
+      get posts_path
+      expect(response).to have_http_status(:ok)
+      expect(response.body).to include('Posts')
+    end
```
Comment (mikethreels) [inline spec/controllers/posts_controller_spec.rb]: - [ ] your spec file should include tests that ensure the correct view template is rendered.
Example : `expect(response).to render_template(:index)`

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Abigail, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @AbigiyaTY,👋🏻👋🏻 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉🚀🔥🎉🎉 


## Highlights ✨✨
- Followed Gitflow ✅
- Great working on the functionality 🔥🚀🎉
- Descriptive PR ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**